### PR TITLE
Improve Zero flag implementation

### PR DESCRIPTION
**Pull request by Tinob**
Originally intended this pull to add support for the rest of the Control flags but I will keep every flag on a separate pull.
This pull improve and centralize the set process of the zero flag

### CODE REVIEW

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 1]
```diff
@@ -1297,7 +1305,7 @@ class GLSLGenerator {
         }
 
         if (dest != Tegra::Shader::Register::ZeroIndex) {
-            regs.SetRegisterToInteger(dest, true, 0, result, 1, 1);
+            regs.SetRegisterToInteger(dest, true, 0, result, 1, 1, false, set_cc);
```
Comment (heapo) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: You've tagged WIP (and if the compiler isn't warning) these parameters are using the old SetRegisterToInteger parameter ordering

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 2]
```diff
@@ -1340,7 +1349,7 @@ class GLSLGenerator {
 
         result += ')';
 
-        regs.SetRegisterToInteger(dest, true, 0, result, 1, 1);
+        regs.SetRegisterToInteger(dest, true, 0, result, 1, 1, false, set_cc);
```
Comment (heapo) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: (here too)

**Comment by heapo** [comment]
Nice approach and great to get rid of some of these unimplementeds

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp (hunk 1 above)
Comment (Tinob) [reply src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: this is using the new paramter layout, set_cc after saturation

Comment (heapo) [reply src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: My mistake, the default parameters threw me off

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 3]
```diff
@@ -404,10 +404,17 @@ class GLSLRegisterManager {
      */
     void SetRegisterToFloat(const Register& reg, u64 elem, const std::string& value,
                             u64 dest_num_components, u64 value_num_components,
-                            bool is_saturated = false, u64 dest_elem = 0, bool precise = false) {
+                            bool is_saturated = false, bool sets_cc = false, u64 dest_elem = 0,
+                            bool precise = false) {
 
         SetRegister(reg, elem, is_saturated ? "clamp(" + value + ", 0.0, 1.0)" : value,
                     dest_num_components, value_num_components, dest_elem, precise);
+        if (sets_cc) {
```
Comment (Subv) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: This will lead to incorrect behavior if RZ is used as the destination register.

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 4]
```diff
@@ -434,7 +441,8 @@ class GLSLRegisterManager {
                     dest_num_components, value_num_components, dest_elem, false);
 
         if (sets_cc) {
```
Comment (Subv) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: Same here

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp (hunk 3 above)
Comment (Tinob) [reply src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: What do you think is the right behavior here? If Rz is the destination the state should be updated?

Comment (Subv) [reply src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: Yes, I've seen some shaders use `I2I.C RZ, R1` or similar, which basically just sets the flags based on R1. This would not work with the implementation in this PR.

Comment (Tinob) [reply src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: should be solved now

**Comment by bunnei** [comment]
@Subv ping to review

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 5]
```diff
@@ -469,10 +492,10 @@ class GLSLRegisterManager {
                 // pack. I couldn't test this on hardware but it shouldn't really matter since most
                 // of the time when a Mrg_* flag is used both components will be mirrored. That
                 // being said, it deserves a test.
-                return "((" + GetRegisterAsInteger(reg, 0, false) +
+                return "uintBitsToFloat((" + GetRegisterAsInteger(reg, 0, false) +
```
Comment (Subv) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: What's up with this seemingly unrelated change?
